### PR TITLE
fix: remote files are not synced

### LINKED ISSUE #1306: Apps won't sync with each other (Ipad and linux)
### Before submitting an issue, please check the following

- [X] I have searched for similar issues (both open and closed) and cannot find a duplicate
- [X] I agree to follow the [Code of Conduct](https://github.com/saber-notes/saber/blob/main/CODE_OF_CONDUCT.md)

### Describe the bug

I have both the linux version (flatpak) and the ipad os version
i'm using my own nextcloud server, and i can see files in the saber dir:
image
When i create a new file it happers in the nextcloud folder, but i can't see it in other clients eg: files created in the ipad app won't appear in the linux app and viceversa:
Linux app:
image
ipad app:
Screenshot 2024-07-08 (00 12 17) jpeg

i've double checked and i'm using the same encryption password in both instances.

### To reproduce

1. create note on device 1
2. create note on device 2
3. See them appear on the nextcloud folder (they can successfully sync)
4. note that was created on device 1 won't appear on device 2 and viceversa

### Expected behavior

Beeing able to see and edit notes from both devices

### Saber version

Linux:
0.24.1
Ios:
0.24.1 (AppStore)

### Device
- Device 1 : Ipad 10
- OS: IpadOS 17.5.1

- Device 2 : my main desktop
- OS: Linux Mint 21.3

### PR DESCRIPTION
**Pull request by QubaB**
fix problems with syncing notes from nextcloud server

closes #1306

### CODE REVIEW

**Comment by QubaB** [comment]
closes  #1304 also

**Comment by adil192** [comment]
Thanks for finding this! I've not been able to for ages.
However, this is just hiding the problem which is that the argument to `FileManager.getFile` should be the local relative path not an absolute path. I'm going to address the root issue in a separate commit and give you credit.

File: lib/data/nextcloud/saber_syncer.dart [hunk 1]
```diff
@@ -78,10 +78,10 @@ class SaberSyncInterface
       final bestFile = await getBestFile(syncFile);
       switch (bestFile) {
         case BestFile.local:
-        case BestFile.either:
           // Local file is newer, do nothing
           break;
         case BestFile.remote:
+        case BestFile.either: //local file does not exists
```
Comment (adil192) [inline lib/data/nextcloud/saber_syncer.dart:84]: Why did you change this part? `BestFile.either` should be when the local and remote files are equivalent, so no sync needs to occur for this file

Comment (QubaB) [reply lib/data/nextcloud/saber_syncer.dart:84]: BestFile.either is also returned when local file does not exist. I did not searched the code why.

Comment (adil192) [reply lib/data/nextcloud/saber_syncer.dart:84]: Thanks, I've improved this aspect in 01b28d9fb2f62672849de58841c573ebad908443
